data_generator.py

```diff
@@ -231,6 +233,20 @@ def __getitem__(self, idx):
                 img, ann['bbox'])
             transformed_label = self.transform_label(
                 ann['keypoints'], cropped_width, cropped_height, anchor_x, anchor_y)
+            if(self.is_eval):
+                metadata = {}
+                metadata["src_set_image_id"] = ann['src_set_image_id']
+                metadata["ann_id"] = ann['ann_id']
+                metadata["coco_url"] = ann['coco_url']
+                metadata["cropped_width"] = cropped_width
+                metadata["cropped_height"] = cropped_height
+                metadata["anchor_x"] = anchor_x
+                metadata["anchor_y"] = anchor_y
+                metadata["input_dim"] = self.input_dim
+                metadata["output_dim"] = self.output_dim
+                metadata["transformed_label"] = transformed_label #DEBUG
+                metadata["ground_truth_keypoints"] = ann['keypoints'] #DEBUG
```

Comment:
Use dictionary constructor so it's easier to read
```suggestion
                metadata = {
                    "src_set_image_id":         ann['src_set_image_id',
                    "ann_id":                   ann['ann_id'],
                    "coco_url":                 ann['coco_url'],
                    "cropped_width":            cropped_width,
                    "cropped_height":           cropped_height,
                    "anchor_x":                 anchor_x,
                    "anchor_y":                 anchor_y,
                    "input_dim":                self.input_dim,
                    "output_dim":               self.output_dim,
                    "transformed_label":        transformed_label, #DEBUG
                    "ground_truth_keypoints":   ann['keypoints'] #DEBUG
                }
```

Reply:
Using that notation is returning a syntax error and I'm not entirely sure why. I have chosen to keep it as is for the time being.

Reply:
Oh it's because the above had a typo on the first line. my bad.
```
ann['src_set_image_id'
```
should be
```
ann['src_set_image_id'],
```

Reply:
I just tested and it no longer gives syntax error.
```suggestion
                metadata = {
                    "src_set_image_id":         ann['src_set_image_id'],
                    "ann_id":                   ann['ann_id'],
                    "coco_url":                 ann['coco_url'],
                    "cropped_width":            cropped_width,
                    "cropped_height":           cropped_height,
                    "anchor_x":                 anchor_x,
                    "anchor_y":                 anchor_y,
                    "input_dim":                self.input_dim,
                    "output_dim":               self.output_dim,
                    "transformed_label":        transformed_label, #DEBUG
                    "ground_truth_keypoints":   ann['keypoints'] #DEBUG
                }
```

Reply:
Good catch! I should have caught that too